nmrfx-processor-gui/src/main/java/org/nmrfx/processor/gui/PolyChart.java

```diff
@@ -2514,6 +2554,55 @@ boolean drawDatasets(GraphicsContextInterface gC) throws GraphicsIOException {
 
     }
 
+
+    /**
+     * Check whether the axis of the dataset attributes are compatible with the first element of the provided attributes
+     * list. Incompatible attributes are removed from the list and the range is adjusted based on the remaining
+     * attributes.
+     * @param attributes The attributes list to remove incompatible datasets from.
+     */
+    private void removeIncompatibleDatasetAttributes(List<DatasetAttributes> attributes) {
+        if (attributes.size() < 2) {
+            return;
+        }
+        int[] matchedDims;
+        Iterator<DatasetAttributes> attributesIterator = attributes.iterator();
+        DatasetAttributes firstAttr = attributesIterator.next();
+        while (attributesIterator.hasNext()) {
+            DatasetAttributes datasetAttributes = attributesIterator.next();
+            matchedDims = firstAttr.getMatchDim(datasetAttributes, true);
+            if (matchedDims[firstAttr.getDim(0)] == -1 || (matchedDims.length > 1 && matchedDims[firstAttr.getDim(1)] == -1)) {
```

Comment:
if (matchedDims[0] == -1 || (matchedDims.length > 1 && matchedDims[1] == -1)) {

With this PR you can no longer display 2D dataset on some 3D datasets.  For example, an hsqc with dimension labels 1H and 15N will not display on an hnco with labels 1H,13C,15N (when displayed with 1H on x axis and 15N on y axis)

Reply:
Changed to not use the getMatchDim method, since it doesn't work well when the axis were switched, now check the first attributes nucleus names and compare, was able to drag the hnco onto the hsqc, change the axis to 1H and 15N and both show up (more visible when the plane is adjusted to centre)